src/Process/Builder/SubscriberBuilder.php

```diff
@@ -130,40 +128,41 @@ public function registerSubscribers(
     private function getSubscribers(
         TestFrameworkAdapter $testFrameworkAdapter,
         OutputInterface $output
-    ): array {
-        $subscribers = [
-            $this->getInitialTestsConsoleLoggerSubscriber($testFrameworkAdapter, $output),
-            $this->getMutantGeneratingConsoleLoggerSubscriber($output),
-            $this->getMutantCreatingConsoleLoggerSubscriber($output),
-            new MutationTestingConsoleLoggerSubscriber(
-                $output,
-                $this->getOutputFormatter($output),
-                $this->metricsCalculator,
-                $this->diffColorizer,
-                $this->showMutations
-            ),
-            new MutationTestingResultsLoggerSubscriber(
-                $this->loggerFactory->createFromLogEntries(
-                    $this->infectionConfig->getLogs(),
-                    $output
-                )
-            ),
-            new PerformanceLoggerSubscriber(
-                $this->stopwatch,
-                $this->timeFormatter,
-                $this->memoryFormatter,
+    ): iterable {
+        yield $this->getInitialTestsConsoleLoggerSubscriber($testFrameworkAdapter, $output);
```

Comment:
why are the subscriber now yielded?

Reply:
Why do we need to buffer then in an array?